### PR TITLE
refactor!: remove key-assignment replacements

### PR DESCRIPTION
**Pull request by insumity**
## Description

Closes: #1664 

Note that we cannot deprecate the `KeyAssignmentReplacementsBytePrefix` because this would change the value of other byte prefixes that [follow](https://github.com/cosmos/interchain-security/blob/v4.0.0/x/ccv/provider/types/keys.go#L128). We therefore just add a `[DEPRECATED]` comment similarly to what is done for [`PendingDataPacketsByteKey`](https://github.com/cosmos/interchain-security/blob/v4.0.0/x/ccv/consumer/types/keys.go#L56) (thanks to @sainoe for pointing this out).

Additionally note that this PR does not introduce any migration to delete any stored `KeyAssignmentReplacement`s. During an upgrade, there would be no `KeyAssignmentReplacement`s stored because in the block before the upgrade, any stored `KeyAssignmentReplacement`s would get deleted when we [queue](https://github.com/cosmos/interchain-security/blob/v4.0.0/x/ccv/provider/keeper/relay.go#L213) the `VSCPacket`s that would call [MustApplyKeyAssignmentToValUpdates](https://github.com/cosmos/interchain-security/blob/main/x/ccv/provider/keeper/key_assignment.go#L563) that deletes all `KeyAssignmentReplacement`s.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

**Comment by insumity** [comment]
> Need s a migration to remove already stored replacements?

@MSalopek Wrote a paragraph in the PR description "Additionally note that this PR ... " on why we do not need a migration. Let me know if you have any concerns.

File: x/ccv/provider/types/keys.go [hunk 1]
```diff
@@ -120,7 +120,9 @@ const (
 	// on consumer chains to validator addresses on the provider chain
```
Comment (sainoe) [inline x/ccv/provider/types/keys.go:120]: nit: the  `KeyAssignmentReplacementsKey` func can be safely removed too.

**Comment by sainoe** [comment]
LGTM! Agree that no migrations are needed imo.

**Review by sainoe — APPROVED** [review]

**Review by p-offtermatt — APPROVED** [review]
LGTM - nice to get rid of this!

**Review by mpoke — APPROVED** [review]
Nice. Removing code is awesome :)

**Pull request merged**